### PR TITLE
OSSM-2153: adds maistra-2.4 presubmit jobs

### PR DESCRIPTION
**Pull request by bartoszmajsak**

### CODE REVIEW

File: prow/config.gen.yaml [hunk 1]
```diff
@@ -2025,6 +2024,310 @@ presubmits:
...
+              value: "0"
+            - name: INTEGRATION_TEST_FLAGS
+              value: ' --istio.test.retries=1 '
+          # FIXME: Uncomment this when we have our proxy in place
+          # - name: ISTIO_ENVOY_BASE_URL
+          #   value: https://storage.googleapis.com/maistra-prow-testing/proxy
+
+          # FIXME: Using upstream for now, replace with our image
+          # image: "quay.io/maistra-dev/maistra-builder:2.4"
+          image: "gcr.io/istio-testing/build-tools:release-1.16-034c1253b7ce447805ce0e76d39a41ab8e466fea"
```
Comment (bartoszmajsak) [inline prow/config.gen.yaml:2056]: @jwendell Is this workaround still relevant? I took it from 2.3 jobs and changed the tag to 1.16 release but I don't know if this what we still need.

Comment (bartoszmajsak) [inline prow/config.gen.yaml:2056]:
```suggestion
          image: "quay.io/maistra-dev/maistra-builder:2.4
```

File: prow/config.gen.yaml [hunk 2]
```diff
@@ -2025,6 +2024,310 @@ presubmits:
...
+            #   value: -mod=vendor
+            - name: BUILD_WITH_CONTAINER
+              value: "0"
+            - name: INTEGRATION_TEST_FLAGS
+              value: ' --istio.test.retries=1 '
+          # FIXME: Uncomment this when we have our proxy in place
+          # - name: ISTIO_ENVOY_BASE_URL
+          #   value: https://storage.googleapis.com/maistra-prow-testing/proxy
+
+          # FIXME: Using upstream for now, replace with our image
```
Comment (jwendell) [inline prow/config.gen.yaml:2054]: nit: remove these comments

File: prow/config/postsubmits.yaml [hunk 3]
```diff
@@ -879,7 +879,6 @@ postsubmits:
           requests:
             cpu: "2"
             memory: 1Gi
-
```
Comment (jwendell) [inline prow/config/postsubmits.yaml:882]: nit: remove this change

File: prow/config/presubmits.yaml [hunk 4]
```diff
@@ -809,6 +809,310 @@ presubmits:
...
+    decorate: true
+    path_alias: istio.io/istio
+    spec:
+      containers:
+        - command:
+            - entrypoint
+            - prow/integ-suite-kind.sh
+            - test.integration.helm.kube.presubmit
+          env:
+            # - name: GOFLAGS
```
Comment (jwendell) [inline prow/config/presubmits.yaml:829]: nit: remove these comments

**Review by jwendell — APPROVED** [review]
LGTM, just minor nits

File: prow/config.gen.yaml (hunk 2 above)
Comment (bartoszmajsak) [reply prow/config.gen.yaml]: It's already gone

File: prow/config/presubmits.yaml (hunk 4 above)
Comment (bartoszmajsak) [reply prow/config/presubmits.yaml:829]: Don't we want to bring it back once applying our changes?

File: prow/config/postsubmits.yaml (hunk 3 above)
Comment (bartoszmajsak) [reply prow/config/postsubmits.yaml]: already done

File: prow/config/presubmits.yaml (hunk 4 above)
Comment (jwendell) [reply prow/config/presubmits.yaml:829]: I don't think so. We are going to experiment removing this entirely for 2.4. Go is smart enough to detect when there's a vendor dir. If anything goes wrong, we can add it back later.

Comment (bartoszmajsak) [reply prow/config/presubmits.yaml]: Ok didn't know that. Removed in 97240a5

**Pull request merged**